third_party/2and3/sqlalchemy/engine/base.pyi

```diff
@@ -1,20 +1,28 @@
-from typing import Any, List, Tuple
+from typing import Any, Iterable, Iterator, List, Mapping, Optional, Tuple
 
 # Dummy until I figure out something better.
 class Connectable:
-    pass
+    def execute(self, object, *multiparams, **params) -> ResultProxy: ...
 
-class Connection:
+class Connection(Connectable):
     def begin(self): ...
-    def execute(self, object, *multiparams, **params): ...
 
-class Engine(object): ...
+class Engine(Connectable): ...
+
+class RowProxy(Mapping[str, Any]):
+    pass
 
-class RowProxy:
-    def items(self) -> List[Tuple[Any, Any]]: ...
-    def keys(self) -> List[Any]: ...
-    def values(self) -> List[Any]: ...
-    def __getitem__(self, key: str): ...
+class ResultProxy(Iterator[RowProxy]):
+    def keys(self) -> List[str]: ...
+    def close(self) -> None: ...
+    def __iter__(self) -> Iterator[RowProxy]: ...
+    def fetchall(self) -> Iterator[RowProxy]: ...
+    def fetchmany(self, size: int=None) -> Iterator[RowProxy]: ...
+    def fetchone(self) -> Optional[RowProxy]: ...
+    def first(self) -> Optional[RowProxy]: ...
+    def scalar(self): ...
```

Comment:
I think returning Any explicitly with a comment is better because it shows that we've thought about it. (I'm hoping we can disallow missing types in typeshed at some point.)